docs/operating/compactor-backlog.md

```diff
@@ -0,0 +1,55 @@
+# Troubleshoot Compactor Backlog
+
+Compactor is one of the most important components in Thanos. It is responsible for doing compaction, downsampling and retention for the data in the object storage.
+
+When your system contains a lot of block producers (Sidecar, Rule, Receiver, etc) or the scale is large, compactor might not be able to keep up with the data producing rate and it falls behind, which causes a lot of backlogged work. This document will help you to troubleshoot the backlog compaction issue and how to scale the compactor.
+
+## Detect the backlog
+
+Self monitoring for the monitoring system is important. We highly recommend you to set up the Thanos Grafana dashboards and alerts to monitor the thanos components. Without self monitoring, it is hard to detect the issue and fix the problems.
+
+If you find these issues in your own Thanos deployment, then your compactor might be backlogged and needs to scale more:
+1. The long-term query in your Grafana dashboard is much slower than usual.
+2. The compactor stops downsampling blocks and `thanos_compact_downsample_total` metric doesn't increase.
+3. The compactor stops retention. If the retention period is set to 1 month but you can still see older blocks from the bucket UI then that's the case.
+4. `thanos_compact_iterations_total` metric doesn't increase or the rate is very low.
+
+Current implementation, compactor will perform compaction, downsampling and retention phases in order, which means that if the compaction work is not finished, downsampling and retention phase won't start. So that's why you will find the symptom 2 and 3 mentioned above.
+
+For symptom 4, `thanos_compact_iterations_total` metric doesn't increase means that the Thanos compactor is currently working on the current compaction iteration and cannot finish it in a long time. It is very similar to the case of a message queue. The producers are the components who upload blocks to your object storage. And the compactor is the consumer to deal with the jobs from producers. If the data produce rate is higher than the processing rate of the consumer, then the compactor will fall behind.
+
+### Compactor progress metrics
+
+Since Thanos v0.24 release, four new metrics `thanos_compact_todo_compactions`, `thanos_compact_todo_compaction_blocks`, `thanos_compact_todo_downsample_blocks` and `thanos_compact_todo_deletion_blocks` are added to show the compaction, downsampling and retention progress and backlog.
```

Comment:
```suggestion
Since the Thanos v0.24 release, four new metrics `thanos_compact_todo_compactions`, `thanos_compact_todo_compaction_blocks`, `thanos_compact_todo_downsample_blocks` and `thanos_compact_todo_deletion_blocks` are added to show the compaction, downsampling and retention progress and backlog.
```